typed-racket-doc/typed-racket/scribblings/reference/special-forms.scrbl

```diff
@@ -614,7 +614,13 @@ protect higher-order uses of the value.
 @ex[
 ((cast (lambda ([s : String]) s) (Any -> Any)) "hello")
 (eval:error ((cast (lambda ([s : String]) s) (Any -> Any)) 5))
-]}
+]
+
+@racket[cast] is mainly useful in the cases in which a thorough check is
+required, in particular when using higher-order or mutable values that cannot
+be fully verified by a flat predicate, or when a good error message from the
+contract system is worth the performance penalty.  In most other cases,
+occurrence typing, @racket[assert], or @racket[ann] should be preferred.}
```

Comment:
I find it a bit odd to mention `ann` here, since it's doing something very different. But I'm open to it depending on what you think.

Reply:
I put it in because it was mentioned in the discussion on Racket Discourse :smile: From my experience, `ann` is indeed doing something rather unrelated, so I would rather agree to remove the mention of `ann` here.